ts/components/session/LeftPaneMessageSection.tsx

```diff
@@ -201,9 +213,78 @@ export class LeftPaneMessageSection extends React.Component<Props, State> {
     );
   }
 
-  private renderClosableOverlay(overlay: SessionComposeToType) {
+  private handleMessageRequestsClick() {
+    this.handleToggleOverlay(SessionClosableOverlayType.MessageRequests);
+  }
+
+  /**
+   * Blocks all message request conversations and synchronizes across linked devices
+   * @returns void
+   */
+  private async handleBlockAllRequestsClick() {
+    let messageRequestsEnabled = false;
+    if (window?.inboxStore?.getState()) {
+      messageRequestsEnabled =
+        window.inboxStore?.getState().userConfig.messageRequests === true &&
+        window.lokiFeatureFlags?.useMessageRequests === true;
+    }
+    if (!messageRequestsEnabled) {
+      return;
+    }
+
+    // block all convo requests. Force sync if there were changes.
+    window?.log?.info('Blocking all conversations');
+    const conversations = getConversationController().getConversations();
+
+    if (!conversations) {
+      window?.log?.info('No message requests to block.');
+      return;
+    }
+
+    const conversationRequests = conversations.filter(conversation => {
+      // Add Open Group to list as soon as the name has been set
+      if (
+        conversation.isPublic() &&
+        (!conversation.get('name') || conversation.get('name') === 'Unknown group')
+      ) {
+        return false;
+      }
+
+      // Remove all invalid conversations and conversatons of devices associated
+      //  with cancelled attempted links
+      if (!conversation.isPublic && !conversation.get('active_at')) {
+        return false;
+      }
+
+      if (conversation.attributes.isApproved || !conversation.get('active_at')) {
+        return false;
+      }
+
+      return true;
+    });
+
```

Comment:
you got to clean this up.
Most of this doesn't make any sense.

if conversation.isPublic() you can drop it completely, no need for check of active_at or name
actually `if !conversation.isPrivate()` you can `return false` right away

I think you only need `!conversation.isPrivate() || !conversation.get('active_at') || conversation.get('isApproved')` to `return false`, otherwise `return true`, right?

this can probably be a single line of 
```ts
const conversationRequests = conversations.filter(c => c.isPrivate() && c.get('active_at') && c.get('isApproved');

```

also, be consistent. If using `.get` syntax, don't use the `.attributes` syntax